---
title: C28205
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28205
helpviewer_keywords:
- C28205
ms.assetid: 3d802885-bdb8-47cb-9108-527a328a0774
ms.openlocfilehash: 18c97b7185628e8b736651dc0ea902216e23b769
ms.sourcegitcommit: 7bea0420d0e476287641edeb33a9d5689a98cb98
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 02/17/2020
ms.locfileid: "77416460"
---
# <a name="c28205"></a>C28205

> aviso C28205: função >: `_Success_` ou `_On_failure_` usado em um contexto inválido

As anotações `_Success_` e `_On_failure_` só podem ser usadas em valores de retorno de função.

## <a name="examples"></a>Exemplos

```cpp
#include<sal.h>

// Oops, _Success_ is not valid in parameter lists, should be moved to return value.
bool GetValue( _Success_(return != false) _Out_ int *pInt, bool flag)
{
   if(flag) {
      *pInt = 5;
      return true;
   } else {
      return false;
   }
}
```

```cpp
#include<sal.h>

_Success_(return != false)
bool GetValue(_Out_ int *pInt, bool flag)
{
   if(flag) {
      *pInt = 5;
      return true;
   } else {
      return false;
   }
}
```

[Compreendendo sal](understanding-sal.md)
